src/components/MoneyRequestConfirmationList.tsx

```diff
@@ -332,19 +332,14 @@ function MoneyRequestConfirmationList({
 
     const shouldShowTax = isTaxTrackingEnabled(isPolicyExpenseChat, policy, isDistanceRequest, isPerDiemRequest);
 
+    // Update the tax code when the default changes (for example, because the transaction currency changed)
+    const defaultTaxCode = getDefaultTaxCode(policy, transaction) ?? '';
```

Comment:
Just in case you were worried, the React Compiler is working in this file, so this will be automatically memoized:

<img width="374" height="136" alt="Image" src="https://github.com/user-attachments/assets/0cf24657-69ac-41ee-bd08-1280435cb55d" />

You can tell by the ✨